### ISSUE TITLE
0x00 byte (%00) in a data: URL

### ISSUE THREAD

**Issue by annevk** [whatwg/fetch#1748]
Briefly looking into https://bugs.webkit.org/show_bug.cgi?id=21975 I noticed that user agents vary in behavior for these two `data:` URLs:

1. `data:text/html,a%00a`
2. `data:text/html,a%01a`

In 1 it seems the 0x00 byte gets dropped completely.

This seems specific to `data:` URLs processing as `%00` remains part of the URL. And for fragment navigation for instance it's not dropped, compare `https://whatwg.org/stages#stage%000` and `https://whatwg.org/stages#stage0`.

**Comment by nektro** [comment]
should this issue be in https://github.com/whatwg/url since this pertains to the https://url.spec.whatwg.org/#percent-encoded-bytes algorithm?

**Comment by annevk** [comment]
@nektro I don't think so. As I noted in my last paragraph this seems specific to `data:` URLs, which are defined in Fetch.
